File: backend/api/getSongs.py
import spotipy
from lyricsgenius import Genius
from spotipy.oauth2 import SpotifyClientCredentials
from . import information

#configure python
client_credentials_manager = \
    SpotifyClientCredentials(client_id=information.ClientID, client_secret=information.ClientSecret)
sp = spotipy.Spotify(client_credentials_manager = client_credentials_manager)

#configure genius
genius = Genius(information.GeniusAccessToken)

def getArtistSong(id, token):
    newOrderURI = "spotify:artist:" + id
    artistName = sp.artist(newOrderURI)["name"]

    # find every album of artist
    results = sp.artist_albums(newOrderURI, album_type="album")
    albums = results["items"]
    while results['next']:
        results = sp.next(results)
        albums.extend(results["items"])

    newAlbums = []
    for album in albums:
        setUpAlbum = {"name": album["name"],"uri":album["uri"],"songs":[]}

        #find every song in album
        results = sp.album_tracks(album["uri"])
        tracks = results["items"]
        while results['next']:
            results = sp.next(results)
            tracks.extend(results["items"])

        newTracks = []
        for track in tracks:
            newTracks.append({"track": track["name"], "uri": track["uri"], "lyrics": ""})

            # find every lyrics to song
            # song = genius.search_song(track["name"], artistName)
            # print(song.lyrics)

        setUpAlbum["songs"] = newTracks
        newAlbums.append(setUpAlbum)

    # print(newAlbums)

File: frontend/src/api/getSongs.js
import { SONGS_ENDPOINT } from "../Information";
import axios from "axios";
import { getLyricsGenius } from "./getLyricsGenius";

const getSongs = async (album, token, artistName) => {
  // get information about songs on album
  return axios.get(SONGS_ENDPOINT(album["id"]), {
    headers: {
      Authorization: "Bearer " + token,
    },
  });
  // const data = request.data.items;
  // //loop on every song in album
  // for (let k = 0; k < data.length; k++) {
  //   var song = {};
  //   song["name"] = data[k].name;
  //   song["id"] = data[k].id;
  //   song["href"] = data[k].href;
  //   song["track_number"] = data[k].track_number;

  //   getLyricsGenius(artistName, album, song);
  //   songs.push(song);
  // }
  // album["songs"] = songs;
  // return songs;
};

export { getSongs };
